### PR TITLE
🌱 Bump golangci to v1.60.2

### LINKED ISSUE #11075: Bump golangci-lint to v1.60.2
Let's bump golangci-lint to the latest v1.60.2 release.

Tasks:
* Bump the linter (prior art: https://github.com/kubernetes-sigs/cluster-api/pull/10398/files)
* Run `make lint` and fix new findings

/help

### PR DESCRIPTION
**Pull request by mcbenjemaa**
**What this PR does / why we need it**:
This PR bumps golang-ci to  v1.60.2,
a lot of false positive issues have appeared, therefore i deal with them.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #11075

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -4,11 +4,6 @@ run:
   build-tags:
     - tools
     - e2e
```
Comment (sbueringer) [inline .golangci.yml:6]: Q: Why was this moved from here to exclude-files? New config file format in 1.60.2?

File: .golangci.yml [hunk 2]
```diff
@@ -62,6 +57,12 @@ linters:
   - whitespace
 
 linters-settings:
+  govet:
+    disable:
+      - printf
+  gosec:
+    excludes:
+      - G115
```
Comment (sbueringer) [inline .golangci.yml:65]: Let's add a comment that this is e.g. " integer overflow conversion int -> int32"

File: .golangci.yml [hunk 3]
```diff
@@ -62,6 +57,12 @@ linters:
   - whitespace
 
 linters-settings:
+  govet:
+    disable:
+      - printf
```
Comment (sbueringer) [inline .golangci.yml:62]: Let's instead use this more precise exclude (via exclude-rules below)
```yaml
  # Ignore non-constant format string in call to condition utils
  - linters:
      - govet
    text: "non-constant format string in call to sigs\\.k8s\\.io\\/cluster-api\\/util\\/conditions\\."
```

(+ then fix the one remaining finding)

File: .golangci.yml (hunk 1 above)
Comment (sbueringer) [reply .golangci.yml:6]: (mostly wondering what the difference is between configuration here or in exclude-files otherwise)

Comment (mcbenjemaa) [reply .golangci.yml:6]: I got a warning `skip-files` is deprecated

Comment (sbueringer) [reply .golangci.yml:6]: Thx, good enough :)

**Comment by mcbenjemaa** [comment]
I also had this linter issues by gosec:


```
exp/internal/webhooks/machinepool.go:255:16: G115: integer overflow conversion int64 -> int32 (gosec)
                        return int32(minSize), nil
                                    ^
exp/internal/webhooks/machinepool.go:269:16: G115: integer overflow conversion int64 -> int32 (gosec)
                        return int32(minSize), nil
                                    ^
exp/internal/webhooks/machinepool.go:272:36: G115: integer overflow conversion int64 -> int32 (gosec)
                case *oldMP.Spec.Replicas < int32(minSize):
                                                 ^
exp/internal/webhooks/machinepool.go:276:16: G115: integer overflow conversion int64 -> int32 (gosec)
                        return int32(minSize), nil
                                    ^
exp/internal/webhooks/machinepool.go:279:36: G115: integer overflow conversion int64 -> int32 (gosec)
                case *oldMP.Spec.Replicas > int32(maxSize):
                                                 ^
exp/internal/webhooks/machinepool.go:283:16: G115: integer overflow conversion int64 -> int32 (gosec)
                        return int32(maxSize), nil
                                    ^
internal/controllers/machinedeployment/mdutil/util.go:187:14: G115: integer overflow conversion int64 -> int32 (gosec)
        return int32(intValue), true
                    ^
internal/controllers/machinedeployment/mdutil/util.go:576:58: G115: integer overflow conversion int -> int32 (gosec)
                maxTotalMachines := *(deployment.Spec.Replicas) + int32(maxSurge)
                                                                       ^
internal/controllers/machinedeployment/mdutil/util.go:616:8: G115: integer overflow conversion int64 -> int32 (gosec)
                int32(desired) == *(deployment.Spec.Replicas) &&
                     ^
internal/controllers/machinedeployment/mdutil/util.go:647:14: G115: integer overflow conversion int -> int32 (gosec)
        return int32(surge), int32(unavailable), nil
                    ^
internal/webhooks/machinedeployment.go:355:16: G115: integer overflow conversion int64 -> int32 (gosec)
                        return int32(minSize), nil
                                    ^
internal/webhooks/machinedeployment.go:369:16: G115: integer overflow conversion int64 -> int32 (gosec)
                        return int32(minSize), nil
                                    ^
internal/webhooks/machinedeployment.go:372:36: G115: integer overflow conversion int64 -> int32 (gosec)
                case *oldMD.Spec.Replicas < int32(minSize):
                                                 ^
internal/webhooks/machinedeployment.go:376:16: G115: integer overflow conversion int64 -> int32 (gosec)
                        return int32(minSize), nil
                                    ^
internal/webhooks/machinedeployment.go:379:36: G115: integer overflow conversion int64 -> int32 (gosec)
                case *oldMD.Spec.Replicas > int32(maxSize):
                                                 ^
internal/webhooks/machinedeployment.go:383:16: G115: integer overflow conversion int64 -> int32 (gosec)
                        return int32(maxSize), nil
                                    ^
internal/webhooks/machineset.go:308:16: G115: integer overflow conversion int64 -> int32 (gosec)
                        return int32(minSize), nil
                                    ^
internal/webhooks/machineset.go:322:16: G115: integer overflow conversion int64 -> int32 (gosec)
                        return int32(minSize), nil
                                    ^
internal/webhooks/machineset.go:325:36: G115: integer overflow conversion int64 -> int32 (gosec)
                case *oldMS.Spec.Replicas < int32(minSize):
                                                 ^
internal/webhooks/machineset.go:329:16: G115: integer overflow conversion int64 -> int32 (gosec)
                        return int32(minSize), nil
                                    ^
internal/webhooks/machineset.go:332:36: G115: integer overflow conversion int64 -> int32 (gosec)
                case *oldMS.Spec.Replicas > int32(maxSize):
                                                 ^
internal/webhooks/machineset.go:336:16: G115: integer overflow conversion int64 -> int32 (gosec)
                        return int32(maxSize), nil
                                    ^
exp/internal/controllers/machinepool_controller_noderef.go:104:33: G115: integer overflow conversion int -> int32 (gosec)
        mp.Status.ReadyReplicas = int32(nodeRefsResult.ready)
                                       ^
exp/internal/controllers/machinepool_controller_noderef.go:105:37: G115: integer overflow conversion int -> int32 (gosec)
        mp.Status.AvailableReplicas = int32(nodeRefsResult.available)
                                           ^
exp/internal/controllers/machinepool_controller_phases_test.go:2089:36: G115: integer overflow conversion int -> int32 (gosec)
                        Replicas:    ptr.To[int32](int32(replicas)),
                                                        ^
internal/controllers/topology/cluster/cluster_controller_test.go:966:11: G115: integer overflow conversion int64 -> int32 (gosec)
                if int32(*replicas) != *cluster.Spec.Topology.ControlPlane.Replicas {
                        ^
internal/controllers/topology/cluster/cluster_controller_test.go:967:67: G115: integer overflow conversion int64 -> int32 (gosec)
                        return fmt.Errorf("replicas %v do not match expected %v", int32(*replicas), *cluster.Spec.Topology.ControlPlane.Replicas)
                                                                                       ^
controlplane/kubeadm/internal/controllers/remediation.go:503:20: G115: integer overflow conversion int -> int32 (gosec)
                RetryCount: int32(r.RetryCount),
                                 ^
controlplane/kubeadm/internal/controllers/status.go:43:49: G115: integer overflow conversion int -> int32 (gosec)
        controlPlane.KCP.Status.UpdatedReplicas = int32(len(upToDateMachines))
                                                       ^
controlplane/kubeadm/internal/controllers/status.go:45:19: G115: integer overflow conversion int -> int32 (gosec)
        replicas := int32(len(controlPlane.Machines))
                         ^
controlplane/kubeadm/internal/controllers/status.go:80:11: G115: integer overflow conversion int -> int32 (gosec)
                if int32(len(readyMachines)) == replicas {
                        ^
controlplane/kubeadm/internal/controllers/upgrade.go:121:54: G115: integer overflow conversion int -> int32 (gosec)
                maxNodes := *controlPlane.KCP.Spec.Replicas + int32(controlPlane.KCP.Spec.RolloutStrategy.RollingUpdate.MaxSurge.IntValue())
                                                                   ^
controlplane/kubeadm/internal/controllers/upgrade.go:122:11: G115: integer overflow conversion int -> int32 (gosec)
                if int32(controlPlane.Machines.Len()) < maxNodes {
                        ^
internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:221:35: G115: integer overflow conversion int -> int32 (gosec)
        m.Status.ExpectedMachines = int32(totalTargets)
                                         ^
internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:236:33: G115: integer overflow conversion int -> int32 (gosec)
        m.Status.CurrentHealthy = int32(len(healthy))
                                       ^
internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:589:27: G115: integer overflow conversion int -> int32 (gosec)
                remediationCount = int32(maxVal - unhealthyMachineCount)
                                        ^
internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:601:26: G115: integer overflow conversion int -> int32 (gosec)
        remediationCount = int32(maxUnhealthy - unhealthyMachineCount)
                                ^
internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:627:12: G115: integer overflow conversion uint64 -> int (gosec)
        return int(minVal), int(maxVal), nil
                  ^
internal/controllers/machinedeployment/machinedeployment_rollout_ondelete.go:122:29: G115: integer overflow conversion int -> int32 (gosec)
                totalMachineCount := int32(len(allMachinesInOldMS.Items))
                                          ^
internal/controllers/machinedeployment/machinedeployment_sync.go:612:15: G115: integer overflow conversion int -> int32 (gosec)
        diff := int32(len(cleanableMSes)) - *deployment.Spec.RevisionHistoryLimit
                     ^
internal/controllers/machineset/machineset_controller.go:952:28: G115: integer overflow conversion int -> int32 (gosec)
        newStatus.Replicas = int32(len(filteredMachines))
                                  ^
internal/controllers/machineset/machineset_controller.go:953:40: G115: integer overflow conversion int -> int32 (gosec)
        newStatus.FullyLabeledReplicas = int32(fullyLabeledReplicasCount)
                                              ^
internal/controllers/machineset/machineset_controller.go:954:33: G115: integer overflow conversion int -> int32 (gosec)
        newStatus.ReadyReplicas = int32(readyReplicasCount)
                                       ^
internal/controllers/machineset/machineset_controller.go:955:37: G115: integer overflow conversion int -> int32 (gosec)
        newStatus.AvailableReplicas = int32(availableReplicasCount)
                                           ^
internal/controllers/machineset/machineset_controller_test.go:1718:49: G115: integer overflow conversion int -> int32 (gosec)
                                                MaxInFlight: ptr.To(intstr.FromInt32(int32(maxInFlight))),
                                                                                          ^
```

Should I do the same?

**Comment by mcbenjemaa** [comment]
> I also had this linter issues by gosec:
> 
> ```
> exp/internal/webhooks/machinepool.go:255:16: G115: integer overflow conversion int64 -> int32 (gosec)
>                         return int32(minSize), nil
>                                     ^
> exp/internal/webhooks/machinepool.go:269:16: G115: integer overflow conversion int64 -> int32 (gosec)
>                         return int32(minSize), nil
>                                     ^
> exp/internal/webhooks/machinepool.go:272:36: G115: integer overflow conversion int64 -> int32 (gosec)
>                 case *oldMP.Spec.Replicas < int32(minSize):
>                                                  ^
> exp/internal/webhooks/machinepool.go:276:16: G115: integer overflow conversion int64 -> int32 (gosec)
>                         return int32(minSize), nil
>                                     ^
> exp/internal/webhooks/machinepool.go:279:36: G115: integer overflow conversion int64 -> int32 (gosec)
>                 case *oldMP.Spec.Replicas > int32(maxSize):
>                                                  ^
> exp/internal/webhooks/machinepool.go:283:16: G115: integer overflow conversion int64 -> int32 (gosec)
>                         return int32(maxSize), nil
>                                     ^
> internal/controllers/machinedeployment/mdutil/util.go:187:14: G115: integer overflow conversion int64 -> int32 (gosec)
>         return int32(intValue), true
>                     ^
> internal/controllers/machinedeployment/mdutil/util.go:576:58: G115: integer overflow conversion int -> int32 (gosec)
>                 maxTotalMachines := *(deployment.Spec.Replicas) + int32(maxSurge)
>                                                                        ^
> internal/controllers/machinedeployment/mdutil/util.go:616:8: G115: integer overflow conversion int64 -> int32 (gosec)
>                 int32(desired) == *(deployment.Spec.Replicas) &&
>                      ^
> internal/controllers/machinedeployment/mdutil/util.go:647:14: G115: integer overflow conversion int -> int32 (gosec)
>         return int32(surge), int32(unavailable), nil
>                     ^
> internal/webhooks/machinedeployment.go:355:16: G115: integer overflow conversion int64 -> int32 (gosec)
>                         return int32(minSize), nil
>                                     ^
> internal/webhooks/machinedeployment.go:369:16: G115: integer overflow conversion int64 -> int32 (gosec)
>                         return int32(minSize), nil
>                                     ^
> internal/webhooks/machinedeployment.go:372:36: G115: integer overflow conversion int64 -> int32 (gosec)
>                 case *oldMD.Spec.Replicas < int32(minSize):
>                                                  ^
> internal/webhooks/machinedeployment.go:376:16: G115: integer overflow conversion int64 -> int32 (gosec)
>                         return int32(minSize), nil
>                                     ^
> internal/webhooks/machinedeployment.go:379:36: G115: integer overflow conversion int64 -> int32 (gosec)
>                 case *oldMD.Spec.Replicas > int32(maxSize):
>                                                  ^
> internal/webhooks/machinedeployment.go:383:16: G115: integer overflow conversion int64 -> int32 (gosec)
>                         return int32(maxSize), nil
>                                     ^
> internal/webhooks/machineset.go:308:16: G115: integer overflow conversion int64 -> int32 (gosec)
>                         return int32(minSize), nil
>                                     ^
> internal/webhooks/machineset.go:322:16: G115: integer overflow conversion int64 -> int32 (gosec)
>                         return int32(minSize), nil
>                                     ^
> internal/webhooks/machineset.go:325:36: G115: integer overflow conversion int64 -> int32 (gosec)
>                 case *oldMS.Spec.Replicas < int32(minSize):
>                                                  ^
> internal/webhooks/machineset.go:329:16: G115: integer overflow conversion int64 -> int32 (gosec)
>                         return int32(minSize), nil
>                                     ^
> internal/webhooks/machineset.go:332:36: G115: integer overflow conversion int64 -> int32 (gosec)
>                 case *oldMS.Spec.Replicas > int32(maxSize):
>                                                  ^
> internal/webhooks/machineset.go:336:16: G115: integer overflow conversion int64 -> int32 (gosec)
>                         return int32(maxSize), nil
>                                     ^
> exp/internal/controllers/machinepool_controller_noderef.go:104:33: G115: integer overflow conversion int -> int32 (gosec)
>         mp.Status.ReadyReplicas = int32(nodeRefsResult.ready)
>                                        ^
> exp/internal/controllers/machinepool_controller_noderef.go:105:37: G115: integer overflow conversion int -> int32 (gosec)
>         mp.Status.AvailableReplicas = int32(nodeRefsResult.available)
>                                            ^
> exp/internal/controllers/machinepool_controller_phases_test.go:2089:36: G115: integer overflow conversion int -> int32 (gosec)
>                         Replicas:    ptr.To[int32](int32(replicas)),
>                                                         ^
> internal/controllers/topology/cluster/cluster_controller_test.go:966:11: G115: integer overflow conversion int64 -> int32 (gosec)
>                 if int32(*replicas) != *cluster.Spec.Topology.ControlPlane.Replicas {
>                         ^
> internal/controllers/topology/cluster/cluster_controller_test.go:967:67: G115: integer overflow conversion int64 -> int32 (gosec)
>                         return fmt.Errorf("replicas %v do not match expected %v", int32(*replicas), *cluster.Spec.Topology.ControlPlane.Replicas)
>                                                                                        ^
> controlplane/kubeadm/internal/controllers/remediation.go:503:20: G115: integer overflow conversion int -> int32 (gosec)
>                 RetryCount: int32(r.RetryCount),
>                                  ^
> controlplane/kubeadm/internal/controllers/status.go:43:49: G115: integer overflow conversion int -> int32 (gosec)
>         controlPlane.KCP.Status.UpdatedReplicas = int32(len(upToDateMachines))
>                                                        ^
> controlplane/kubeadm/internal/controllers/status.go:45:19: G115: integer overflow conversion int -> int32 (gosec)
>         replicas := int32(len(controlPlane.Machines))
>                          ^
> controlplane/kubeadm/internal/controllers/status.go:80:11: G115: integer overflow conversion int -> int32 (gosec)
>                 if int32(len(readyMachines)) == replicas {
>                         ^
> controlplane/kubeadm/internal/controllers/upgrade.go:121:54: G115: integer overflow conversion int -> int32 (gosec)
>                 maxNodes := *controlPlane.KCP.Spec.Replicas + int32(controlPlane.KCP.Spec.RolloutStrategy.RollingUpdate.MaxSurge.IntValue())
>                                                                    ^
> controlplane/kubeadm/internal/controllers/upgrade.go:122:11: G115: integer overflow conversion int -> int32 (gosec)
>                 if int32(controlPlane.Machines.Len()) < maxNodes {
>                         ^
> internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:221:35: G115: integer overflow conversion int -> int32 (gosec)
>         m.Status.ExpectedMachines = int32(totalTargets)
>                                          ^
> internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:236:33: G115: integer overflow conversion int -> int32 (gosec)
>         m.Status.CurrentHealthy = int32(len(healthy))
>                                        ^
> internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:589:27: G115: integer overflow conversion int -> int32 (gosec)
>                 remediationCount = int32(maxVal - unhealthyMachineCount)
>                                         ^
> internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:601:26: G115: integer overflow conversion int -> int32 (gosec)
>         remediationCount = int32(maxUnhealthy - unhealthyMachineCount)
>                                 ^
> internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:627:12: G115: integer overflow conversion uint64 -> int (gosec)
>         return int(minVal), int(maxVal), nil
>                   ^
> internal/controllers/machinedeployment/machinedeployment_rollout_ondelete.go:122:29: G115: integer overflow conversion int -> int32 (gosec)
>                 totalMachineCount := int32(len(allMachinesInOldMS.Items))
>                                           ^
> internal/controllers/machinedeployment/machinedeployment_sync.go:612:15: G115: integer overflow conversion int -> int32 (gosec)
>         diff := int32(len(cleanableMSes)) - *deployment.Spec.RevisionHistoryLimit
>                      ^
> internal/controllers/machineset/machineset_controller.go:952:28: G115: integer overflow conversion int -> int32 (gosec)
>         newStatus.Replicas = int32(len(filteredMachines))
>                                   ^
> internal/controllers/machineset/machineset_controller.go:953:40: G115: integer overflow conversion int -> int32 (gosec)
>         newStatus.FullyLabeledReplicas = int32(fullyLabeledReplicasCount)
>                                               ^
> internal/controllers/machineset/machineset_controller.go:954:33: G115: integer overflow conversion int -> int32 (gosec)
>         newStatus.ReadyReplicas = int32(readyReplicasCount)
>                                        ^
> internal/controllers/machineset/machineset_controller.go:955:37: G115: integer overflow conversion int -> int32 (gosec)
>         newStatus.AvailableReplicas = int32(availableReplicasCount)
>                                            ^
> internal/controllers/machineset/machineset_controller_test.go:1718:49: G115: integer overflow conversion int -> int32 (gosec)
>                                                 MaxInFlight: ptr.To(intstr.FromInt32(int32(maxInFlight))),
>                                                                                           ^
> ```
> 
> Should I do the same?

Alright, i see you're okay with the disable the specific rule.

File: .golangci.yml (hunk 2 above)
Comment (mcbenjemaa) [reply .golangci.yml:63]: done

**Comment by sbueringer** [comment]
@mcbenjemaa Can you please rebase? There was another PR touching the golangci-lint file merged in the meantime

**Comment by sbueringer** [comment]
PR in general lgtm. Just not sure if the merge works correctly because exclude-files was added in another place in the other PR. So I'm not sure if we get a duplicate when we just merge this PR

File: .golangci.yml [hunk 4]
```diff
@@ -348,3 +352,12 @@ issues:
...
     path: _test\.go
+    # Ignore non-constant format string in call to condition utils
+  - linters:
+      - govet
+    text: "non-constant format string in call to sigs\\.k8s\\.io\\/cluster-api\\/util\\/conditions\\."
+  exclude-files:
+    - "zz_generated.*\\.go$"
+    - "vendored_openapi\\.go$"
+    # We don't want to invest time to fix new linter findings in old API types.
+    - "internal/apis/.*"
```
Comment (sbueringer) [inline .golangci.yml:363]: This also exists above. Let's delete on occurence

**Comment by sbueringer** [comment]
Thank you very much!!

/lgtm
/approve
